src/validation_workflow/docker/validation_docker.py

```diff
@@ -0,0 +1,203 @@
+
+            # STEP 3 . spin-up OS/OSD cluster
+            return_code, self._target_yml_file = self.run_container(
+                self.args.OS_image,
+                self.args.OSD_image,
+                self.args.version,
+                'opensearch-node1-test',
+                'opensearch-node2-test',
+                'opensearch-dashboards-test',
+            )
+            if (return_code):
+                logging.info('Cluster is running now')
+                logging.info('Checking if cluster is ready for API test every 10 seconds\n\n')
+                while True:
+                    if self.check_http_request():
+                        logging.info('\n\ncluster is now ready for API test\n\n')
+                        break
+                    logging.info('sleeping 10sec')
+                    time.sleep(10)
+
+                # STEP 4 . OS, OSD API validation
+                _test_result, _counter = TestCases().test_cases()
+
+                if (_test_result):
+                    logging.info(f'All tests Pass : {_counter}')
+                    return True
+                else:
+                    raise Exception(f'Not all tests Pass : {_counter}')
+            else:
+                raise Exception('The container is fail to run. Exiting the validation.')
+        else:
+            logging.info('\n\nImage digest is Not validated. Exiting..\n\n')
+            raise Exception('Image digest does not match between the opensearchstaging at dockerHub and opensearchproject at local download. Exiting the validation.')
+
+    def check_http_request(self) -> bool:
+        try:
+            status_code, response_text = ApiTest('https://localhost:9200/').api_get()
+            if status_code == 200:
+                return True
+            else:
+                return False
+        except (requests.exceptions.ConnectionError, requests.exceptions.ConnectTimeout) as e:
+            logging.error(f'Error connecting to https://localhost:9200/: {e}')
+            return False
+
+    def cleanup(self) -> bool:
+        # clean up
+        if self.cleanup_process():
+            logging.info('cleanup is completed')
+            return True
+        else:
+            logging.info('cleanup is Not completed')
+            return False
+
+    def cleanup_process(self) -> bool:
+        # stop the containers
+        docker_command = f'{"docker-compose -f"} {self._target_yml_file} {"down"}'
+        result = subprocess.run(docker_command, shell=True, stdout=PIPE, stderr=PIPE, universal_newlines=True)
+
+        # remove docker-compose.yml from the tmp folder
+        try:
+            os.remove(self._target_yml_file)
+        except OSError as e:
+            print("Error: %s - %s." % (e.filename, e.strerror))
+
+        return('returncode=0' in (str(result)))
+
+    def is_container_daemon_running(self) -> Any:
+        check_docker_bash = "./src/validation_workflow/docker/check_docker_daemon.sh"
```

Comment:
Use path.join to connect paths everywhere else these will fail on windows

Reply:
Thanks, @gaiksaya , makes sense... better than just using '+' , let us update it at next commit.